### PR TITLE
Separate add/remove product button into two separate buttons

### PR DESCRIPTION
**Pull request by davidmoore1**
This pull request separates the add remove product button into two separate add and remove buttons.  Pressing add shows only the products to be added which are removed from the list as they are added.  Pressing remove shows only the existing products and removes them from the list as they are deselected.  This PR also fixes issues where the underlying list on the project screen was not updated when products were added and deleted.
Project screen:
Screen Shot 2020-01-10 at 10 46 12 AM
Remove product popup:
Screen Shot 2020-01-10 at 10 46 51 AM
Add product popup:
Screen Shot 2020-01-10 at 10 46 40 AM

### CODE REVIEW

File: source/SIL.AppBuilder.Portal.Frontend/src/data/containers/with-relationship.tsx [hunk 1]
```diff
@@ -117,7 +117,10 @@ export function relationsFromPath(
 
   for (let i = 0; i < relationshipPath.length; i++) {
     segment = relationshipPath[i];
-    assert(`can't access segment (${segment}) on falsey model.`, !!model);
+    if (!model) {
+      console.debug(`can't access segment (${segment}) on falsey model.`);
+      return models;
+    }
```
Comment (chrisvire) [inline source/SIL.AppBuilder.Portal.Frontend/src/data/containers/with-relationship.tsx]: What is this about? Did you intend to leave it this way?

File: source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/projects/show/__tests__/products-with-stores-acceptance-test.tsx [hunk 2]
```diff
@@ -282,15 +286,18 @@ describe('Acceptance | Project View | Products With Stores', () => {
             await when(() =>
               assert(stores.isOverlayLoaderVisible, 'expected overlay loader to briefly be visible')
             );
-            await when(() => products.checkedItems().length > 0);
+            await when(() => products.itemNamed('android_s3').isVisible);
```
Comment (chrisvire) [inline source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/projects/show/__tests__/products-with-stores-acceptance-test.tsx]: Is it OK to use this specific name in the test?

**Comment by chrisvire** [comment]
I talked with Bill Dyck about the proposed UI changes.  I want to be careful to not be [bikeshedding](https://en.wiktionary.org/wiki/bikeshedding).  

Here is our feedback and some proposed changes:
1. the checkboxes are not needed anymore and are actually confusing
2. the titles of the dialogs should reflect what we want the user to do:
    a. Click Product to Add
    b. Click Product to Remove
3. the gray Android icon in the Add dialog seems like it is disabled/grayed out
4. the X and the done button are redundant (and the X and the cancel button when setting the store).  Should we do anything? Maybe not. Seeing an X and  Done button makes it seem like the actions happen when you click Done and you can cancel the dialog without having any affect.

**Comment by davidmoore1** [comment]
Dio what do you want in place of checkboxes

**Comment by chrisvire** [comment]
> Dio what do you want in place of checkboxes

I think we would suggest to remove the checkbox from the entry.

**Pull request merged**